presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -0,0 +1,455 @@
+                    }
+                    if (columnFamily.getMaxVersions() != 1) {
+                        properties.put(PhoenixTableProperties.VERSIONS, columnFamily.getMaxVersions());
+                    }
+                    if (columnFamily.getMinVersions() > 0) {
+                        properties.put(PhoenixTableProperties.MIN_VERSIONS, columnFamily.getMinVersions());
+                    }
+                    if (!columnFamily.getCompression().toString().equals("NONE")) {
+                        properties.put(PhoenixTableProperties.COMPRESSION, columnFamily.getCompression().toString());
+                    }
+                    if (columnFamily.getTimeToLive() < FOREVER) {
+                        properties.put(PhoenixTableProperties.TTL, columnFamily.getTimeToLive());
+                    }
+                    break;
+                }
+            }
+        }
+        catch (IOException | SQLException e) {
+            throw new PrestoException(PHOENIX_METADATA_ERROR, "Couldn't get Phoenix table properties", e);
+        }
+        return properties.build();
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName, Map<String, Object> properties)
+    {
+        checkArgument(!DEFAULT_SCHEMA.equalsIgnoreCase(schemaName), "Can't create 'default' schema which maps to Phoenix empty schema");
+        checkArgument(properties.isEmpty(), "Can't have properties for schema creation");
+        phoenixClient.execute(session, format("CREATE SCHEMA %s", getEscapedArgument(toMetadataCasing(session, schemaName))));
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        checkArgument(!DEFAULT_SCHEMA.equalsIgnoreCase(schemaName), "Can't delete 'default' schema which maps to Phoenix empty schema");
+        phoenixClient.execute(session, format("DROP SCHEMA %s", getEscapedArgument(toMetadataCasing(session, schemaName))));
+    }
+
+    private String toMetadataCasing(ConnectorSession session, String schemaName)
+    {
+        try (Connection connection = phoenixClient.getConnection(JdbcIdentity.from(session))) {
+            boolean uppercase = connection.getMetaData().storesUpperCaseIdentifiers();
+            if (uppercase) {
+                schemaName = schemaName.toUpperCase(ENGLISH);
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(PHOENIX_METADATA_ERROR, "Couldn't get casing for the schema name", e);
+        }
+        return schemaName;
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+    {
+        JdbcTableHandle existingTable = getTableHandle(session, tableMetadata.getTable());
+        if (existingTable != null && !ignoreExisting) {
+            throw new IllegalArgumentException("Target table already exists: " + tableMetadata.getTable());
```

Comment:
Throw `PrestoException(ALREADY_EXISTS)` here, since this is a user error. Also, it's fine to unconditionally throw, regardless of `ignoreExisting`. That flag is there for transactional connectors like Hive that create tables later during the commit call. For non-transactional connectors, the engine's `CreateTableTask` will correctly handle this if the connector throws with `ALREADY_EXISTS`.

In practice, this only ever happens in a race condition, since the engine already checks if the table exists before calling this method. And actually, we still have a race condition, since this check happens before the actual create.

So, I would say just remove this check, since it's redundant with the one the engine performs.